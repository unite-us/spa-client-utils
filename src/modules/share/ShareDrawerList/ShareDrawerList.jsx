import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { get } from 'lodash';
import ShareDrawerListItem from '../ShareDrawerListItem';

const ShareDrawerList = ({
  className,
  displayProperty,
  listItems,
  keyProperty,
}) => (
  <ul className={classNames('share-drawer-list', className)}>
    {
      listItems.map(item => (
        <ShareDrawerListItem key={get(item, keyProperty)}>
          {get(item, displayProperty)}
        </ShareDrawerListItem>
      ))
    }
  </ul>
);

ShareDrawerList.propTypes = {
  /** class name */
  className: PropTypes.string,
  /** list item property for display, defaults to 'name' */
  displayProperty: PropTypes.string,
  /** array of list items */
  listItems: PropTypes.arrayOf(PropTypes.object),
  /** list item property used as key, defaults to 'id' */
  keyProperty: PropTypes.string,
};

ShareDrawerList.defaultProps = {
  className: '',
  displayProperty: 'name',
  listItems: [],
  keyProperty: 'id',
};

export default ShareDrawerList;
