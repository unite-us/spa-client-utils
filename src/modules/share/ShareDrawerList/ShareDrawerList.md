```jsx
const listItems = [
  { id: 1, name: 'First List Item' },
  { id: 2, name: 'Second List Item' },
  { id: 3, name: 'Third List Item' },
  { id: 4, name: 'Fourth List Item' },
];
<ShareDrawerList
  className="this-share-drawer-list"
  listItems={listItems}
/>
```
