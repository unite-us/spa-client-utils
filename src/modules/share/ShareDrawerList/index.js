import ShareDrawerList from './ShareDrawerList';

export default ShareDrawerList;
