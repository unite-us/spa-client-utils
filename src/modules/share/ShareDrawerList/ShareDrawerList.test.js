import React from 'react';
import renderer from 'react-test-renderer';
import ShareDrawerList from './ShareDrawerList';

describe('ShareDrawerList', () => {
  it('renders list', () => {
    const listItems = [
      { name: 'One', id: 1 },
      { name: 'Two', id: 2 },
      { name: 'Three', id: 3 },
    ];
    expect(renderer.create(
      <ShareDrawerList
        className="additional-class"
        listItems={listItems}
      />,
    )).toMatchSnapshot();
  });

  it('renders list with custom item properties', () => {
    const listItems = [
      { display: 'One', key: 1 },
      { display: 'Two', key: 2 },
      { display: 'Three', key: 3 },
    ];
    expect(renderer.create(
      <ShareDrawerList
        className="additional-class"
        displayProperty="display"
        listItems={listItems}
        keyProperty="key"
      />,
    )).toMatchSnapshot();
  });
});
