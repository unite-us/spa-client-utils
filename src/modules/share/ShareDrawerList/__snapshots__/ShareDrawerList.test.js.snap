// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareDrawerList renders list 1`] = `
<ul
  className="share-drawer-list additional-class"
>
  <li
    className="share-drawer-list-item"
  >
    One
  </li>
  <li
    className="share-drawer-list-item"
  >
    Two
  </li>
  <li
    className="share-drawer-list-item"
  >
    Three
  </li>
</ul>
`;

exports[`ShareDrawerList renders list with custom item properties 1`] = `
<ul
  className="share-drawer-list additional-class"
>
  <li
    className="share-drawer-list-item"
  >
    One
  </li>
  <li
    className="share-drawer-list-item"
  >
    Two
  </li>
  <li
    className="share-drawer-list-item"
  >
    Three
  </li>
</ul>
`;
